File: 05 Material Properties/materials/SolidWorksMacro.vb
Imports SolidWorks.Interop.sldworks
Imports SolidWorks.Interop.swconst
Imports System.Runtime.InteropServices
Imports System
Imports System.Globalization
Imports System.Windows.Forms

Partial Class SolidWorksMacro
    Public Sub main()
        'Initialize the dialog
        Dim MyMaterials As New Dialog1
        Dim MyCombo As Windows.Forms.ComboBox
        MyCombo = MyMaterials.MaterialsCombo

        'set up materials list
        Dim MyProps() As String = {"Alloy Steel", "6061 Alloy", "ABS PC"}
        MyCombo.Items.AddRange(MyProps)

        Dim Result As DialogResult
        Result = MyMaterials.ShowDialog()

        If Result = DialogResult.OK Then
            Dim Model As ModelDoc2 = swApp.ActiveDoc
            If Model Is Nothing Then
                MsgBox("You must first open a file.", MsgBoxStyle.Exclamation)
                Exit Sub
            End If

            If Model.GetType = swDocumentTypes_e.swDocPART Then
                'Assign the material to the part
                Dim Part As PartDoc = Model
                'Part = swApp.ActiveDoc
                Part.SetMaterialPropertyName2("Default",
                   "SOLIDWORKS Materials.sldmat", MyCombo.Text)
            ElseIf Model.GetType = swDocumentTypes_e.swDocASSEMBLY Then
                Dim Assy As AssemblyDoc = Model
                'set materials on selected components
                Dim SelMgr As SelectionMgr
                SelMgr = Model.SelectionManager
                Dim Comp As Component2
                Dim compModel As ModelDoc2
                If SelMgr.GetSelectedObjectCount2(-1) < 1 Then
                    MsgBox("You must select at least one component.", MsgBoxStyle.Exclamation)
                    Exit Sub
                End If

                For i As Integer = 1 To SelMgr.GetSelectedObjectCount2(-1)
                    Comp = SelMgr.GetSelectedObjectsComponent4(i, -1)
                    compModel = Comp.GetModelDoc2
                    If compModel.GetType = swDocumentTypes_e.swDocPART Then
                        compModel.SetMaterialPropertyName2("Default", "SOLIDWORKS Materials.sldmat", MyCombo.Text)
                    End If
                Next

            End If
        End If




    End Sub
    ''' <summary>
    ''' The SldWorks swApp variable is pre-assigned for you.
    ''' </summary>
    Public swApp As SldWorks
End Class

File: C# Examples/05 Material Properties/materials/SolidWorksMacro.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;
using System.Windows.Forms;

using SolidWorks.Interop.sldworks;
using SolidWorks.Interop.swconst;


namespace materials
{
    public partial class SolidWorksMacro
    {
public void main()
{
    // Initialize the dialog
    var MyMaterials = new Dialog1();
    ComboBox MyCombo;
    MyCombo = MyMaterials.MaterialsCombo;

            // Set up materials list
    var MyProps = new string[] { "Alloy Steel", "6061 Alloy", "ABS PC"};
    MyCombo.Items.AddRange(MyProps);
    DialogResult Result;
    Result = MyMaterials.ShowDialog();
    if (Result == DialogResult.OK)
    {
        ModelDoc2 Model = (ModelDoc2)swApp.ActiveDoc;
        if (Model == null)
        {
            MessageBox.Show("You must first open a file.", 
                "Materials", MessageBoxButtons.OK, MessageBoxIcon.Warning);
            return;
        }

        if ((int)Model.GetType() == (int)swDocumentTypes_e.swDocPART)
        {
            // Assign the material to the part
            PartDoc Part = (PartDoc)Model;
            // Part = swApp.ActiveDoc
            Part.SetMaterialPropertyName2("Default", "SOLIDWORKS Materials.sldmat", MyCombo.Text);
        }
        else if ((int)Model.GetType() == (int)swDocumentTypes_e.swDocASSEMBLY)
        {
            AssemblyDoc Assy = (AssemblyDoc)Model;
            // set materials on selected components
            SelectionMgr SelMgr;
            SelMgr = (SelectionMgr)Model.SelectionManager;
            Component2 Comp;
            ModelDoc2 compModel;
            if (SelMgr.GetSelectedObjectCount2(-1) < 1)
            {
                MessageBox.Show("You must select at least one component.", "Materials", MessageBoxButtons.OK, MessageBoxIcon.Warning);
                return;
            }

            for (int i = 1, loopTo = SelMgr.GetSelectedObjectCount2(-1); i <= loopTo; i++)
            {
                Comp = (Component2)SelMgr.GetSelectedObjectsComponent4(i, -1);
                compModel = (ModelDoc2)Comp.GetModelDoc2();
                if (compModel.GetType() == (int)swDocumentTypes_e.swDocPART)
                {
                    PartDoc swPart = (PartDoc)compModel;
                    swPart.SetMaterialPropertyName2("Default", "SOLIDWORKS Materials.sldmat", MyCombo.Text);
                }
            }
        }
    }
}


        // The SldWorks swApp variable is pre-assigned for you.
        public SldWorks swApp;

    }
}

